cmd/lp-colouring/main.go

```diff
@@ -87,8 +87,8 @@ func LaunchGraphExecution(gName string, async bool, dynamic bool) *graph.Graph[V
 
 	g := &graph.Graph[VertexProperty, EdgeProperty, MessageValue]{}
 	g.SourceInit = false
-	g.InitVal = 0
-	g.EmptyVal = EMPTYVAL
+	g.InitVal = []IdColourPair{{Colour: EMPTYVAL}}
```

Comment:
I think it is worth noting why the "initial value" and "empty value" are non-empty, as it specifies the colour of Vertex 0. :)

If I understand correctly, it is because the MessageAggregator function assumes VisitMsg contains one element, and the framework has to send the initial value to some vertex, calling MessageAggregator with the initial value.

Reply:
> as it specifies the colour of Vertex 0.
Well, it was protected against actually writing that or thinking its vertex 0 by the check for emptyval, but you're right this is confusing and its probably better off with just nil's here.

For this algorithm it is initialized with a visit to all vertices, but the value should be meaningless.